whirlpool/oven.py

```diff
@@ -244,6 +247,27 @@ async def set_cook(self, mode: CookMode = CookMode.Bake, target_temp: float = No
         
         await self.send_attributes(attrs)
 
+    async def set_bake(self, target_temp: float = None, cavity: Cavity = Cavity.Upper):
```

Comment:
sorry, but I may have explained myself worng :/ 
the idea was to have 4 methods:
```
set_cook_mode(mode: CookMode)
set_target_temp(temp: float)
set_cook_operation(operation: CookOperation)
set_meat_probe_target_temp(temp: float)
```
a lib client such as HA can then call them as needed. For example, an automation or UI interacation can call set_target_temp(...), but not call set_cook_operation(CookOperation.Start) immediately, for example.

Reply:
My suggestion for the `set_timer` is the same as this

Reply:
Ah, I see.  I can do that, but before I do, I should mention why I've done it this way so you understand -- I think there are tradeoffs with that approach.  You actually cannot directly e.g. set "just" the target temp with at least the oven I have: it's not a valid operation.  You have to make a call that includes several things all bundled together: the target temp, the target mode, and a special flag that "starts" it.  The same is true for the timer (at least from what I've seen).  If you try to send a command that just changes the target temp but does not e.g. also send that the cook mode should be Bake (or Broil or whatever), the oven won't change the target temperature: it will reject as an invalid command.

So if you want these utility functions for e.g. set_target_temp without also providing a cookmode (or in the case of the timer, setting a timer without including the start or cancel timer operation), the code will need to be refactored in 1 of 2 ways to make it operational:
1. Maintain the state of the oven in the class object (e.g. when someone calls set_cook_mode(), persist that cook mode in the object) but that that the additional downside of essentially having race condition when "competing" against a user that may be changing the status outside of the app/class
2. Look up the current cook mode by doing a status update when someone calls e.g. set_target_temp() so we can add the necessary config back in.  This has the additional downside of essentially needing an update call for many/most operations with the oven

Beyond the 2 downsides of each of these above approaches, the reason I chose to do this all in a single step rather than (1) or (2) is these open up additional error conditions -- or at least indeterminate outcomes. e.g. if someone calls set_target_temp and cook_mode hasn't yet been called, that's an invalid operation for the oven, so what should we do for the user?  We could throw an error then (because the oven will) or we can make some assumption(s) (e.g. if someone calls set_target_temp and cook_mode isn't set, make the assumption they were looking for CookMode.Bake)